### ISSUE TITLE
[@types/mongodb] FilterQuery does not type check anything (is practically an object)

### ISSUE THREAD

**Issue by HosseinAgha** [DefinitelyTyped/DefinitelyTyped#37404]
- [X] I tried using the `@types/mongodb` package and had problems.
- [X] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [X] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [X] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @CaselIT @alanmarcell @bitjson

I think the type definitions for `FilterQuery` is fundamentally wrong. They are practically `Object` type.  

```typescript
export type FilterQuery<T> = {
    [P in keyof T]?: T[P] | Condition<T, P>;
} | { [key: string]: any };
```
The `{ [key: string]: any }` completely bypasses any type before it. It is like `string | number | any`.   

I don't know why we put it there? but it makes `FilterQuery` useless. I can practically put anything in my `find` query.   

The same applies to the `Condition` type definition.

I think we should create a failing test for this case e.g. 
```javascript
        interface Bag {
            cost: number;
            color: string;
        }
        const cursor: mongodb.Cursor<Bag> = collection.find<Bag>({ color: 12 });
```

I don't know how to achieve that in the current testing pipeline. Could you please help me?   
How should ask the maintainers of the `DefinitelyTyped` to add option?

**Comment by HosseinAgha** [comment]
OK after some digging I understand that we have this because of the support for MongoDB nested queries. e.g. this is not supported if we remove the `{ [key: string]: any }`:  

```typescript
collection('cars').findOne({ 'size.length': 12 })
```

but I still think this should have not been added. If we cannot support typings we should simply comment this and add a `any`. This is misleading as users think we are supporting the types for query filters.  

As an example for how wrong these definitions are:
this is accepted in typescript but mongo throws a large error.
```
schemas.findOne({ name: 'hello', $and: { a: 23 } })
```

**Comment by CaselIT** [comment]
yes that was the reason.
if you find a better type for it, open a pull request.

>If we cannot support typings we should simply comment this and add a `any`

that could be an option

**Comment by HosseinAgha** [comment]
Answering myself here: To write a failing test we should use [dtslint](https://github.com/Microsoft/dtslint#write-tests) `$ExpectType` and `$ExpectError`.  
Closing as this issue will be fixed by #37467.
